Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensionsTest.kt

```diff
@@ -85,244 +86,206 @@ class PersonCertificatesExtensionsTest : BaseTest() {
             every {
                 rawCertificate.vaccination.vaccinatedOn
             } returns time.minus(Duration.standardDays(25)).toLocalDateUtc()
+            every { rawCertificate.vaccination.medicalProductId } returns "EU/1/20/1528"
             every { getState() } returns mockk<State.Valid>()
         }
-
-        val thirdIssuedAtAnotherDate = mockk<VaccinationCertificate>().apply {
-            every { headerIssuedAt } returns Instant.parse("2021-07-20T10:00:00.000Z")
-            every { rawCertificate.vaccination.doseNumber } returns 2
+        // Vaccination certificate with dose 1/2 and < 14 days since administration
+        val fifth = mockk<VaccinationCertificate>().apply {
+            every { headerIssuedAt } returns time
+            every { rawCertificate.vaccination.doseNumber } returns 1
             every { rawCertificate.vaccination.totalSeriesOfDoses } returns 2
             every {
                 rawCertificate.vaccination.vaccinatedOn
-            } returns time.minus(Duration.standardDays(25)).toLocalDateUtc()
+            } returns time.minus(Duration.standardDays(1)).toLocalDateUtc()
+            every { rawCertificate.vaccination.medicalProductId } returns "EU/1/20/1528"
             every { getState() } returns mockk<State.Valid>()
         }
-
-        val thirdIsBooster = mockk<VaccinationCertificate>().apply {
-            every { headerIssuedAt } returns Instant.parse("2021-11-05T14:00:00.000Z")
-            every { rawCertificate.vaccination.doseNumber } returns 3
-            every { rawCertificate.vaccination.totalSeriesOfDoses } returns 3
+        // Recovery certificate < 180 days old
+        val sixth = mockk<RecoveryCertificate>().apply {
+            every { headerIssuedAt } returns time.minus(Duration.standardDays(10))
+            every { rawCertificate.recovery.validFrom } returns time.minus(Duration.standardDays(10))
+                .toLocalDateUtc()
+            every { rawCertificate.recovery.testedPositiveOn } returns time.minus(Duration.standardDays(30))
+                .toLocalDateUtc()
+            every { getState() } returns mockk<State.Valid>()
+        }
+        // Vaccination certificate of type J&J and < 14 days old
+        val seventh = mockk<VaccinationCertificate>().apply {
+            every { headerIssuedAt } returns time
+            every { rawCertificate.vaccination.doseNumber } returns 1
+            every { rawCertificate.vaccination.totalSeriesOfDoses } returns 1
             every {
                 rawCertificate.vaccination.vaccinatedOn
-            } returns time.minus(Duration.standardDays(15)).toLocalDateUtc()
+            } returns time.minus(Duration.standardDays(1)).toLocalDateUtc()
+            every { rawCertificate.vaccination.medicalProductId } returns "EU/1/20/1525"
             every { getState() } returns mockk<State.Valid>()
         }
-
-        val thirdIsBoosterIssuedAtAnotherDate = mockk<VaccinationCertificate>().apply {
```

Comment:
Are you still covering these cases in the tests? 
It could be that you have the same booster dose ,but issued with in different dates. 
Same dose has two certificates , the two certificates have the same `vaccinatedOn` date , but `issuedAt` is different

Reply:
Sure, I can add one more.